lib/vet360/stats.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module Vet360
+  class Stats
+    FINAL_SUCCESS = %w[COMPLETED_SUCCESS COMPLETED_NO_CHANGES_DETECTED].freeze
+    FINAL_FAILURE = %w[REJECTED COMPLETED_FAILURE].freeze
+
+    class << self
+      def exception_keys
+        exceptions_file
+          .dig('en', 'common', 'exceptions')
+          .keys
+          .select { |exception| exception.include? 'VET360_' }
+          .sort
+          .map(&:downcase)
+      end
+
+      def increment(*args)
+        buckets = args.map(&:downcase).join('.')
+
+        StatsD.increment("#{Vet360::Service::STATSD_KEY_PREFIX}.#{buckets}")
+      end
+
+      def increment_transaction_results(response, bucket1 = 'posts_and_puts')
+        status = status_in(response)
+
+        return unless final_status?(status)
+
+        increment(bucket1, bucket_for(status))
+      end
+
+      private
+
+      def exceptions_file
+        config = Rails.root + 'config/locales/exceptions.en.yml'
+
+        YAML.load_file(config)
```

Comment:
I'm concerned that we're hitting the filesystem for a read on a quite large file on what should be a lightweight stats call. Is there anywhere this exception data is loaded already by rails, given it's a rails idiom?

Reply:
Now that I think about it, the parsing overhead in addition to the cost of reading from the filesystem probably makes this a bit too slow in my opinion. Might be something worth benchmarking as well, if we need the hard figures to make a decision.

Reply:
Dismissing my review, this doesn't apply to the hot path of requests.